### PR TITLE
refactor: fetch on load & stable refetch

### PR DESCRIPTION
**Pull request by BiswaViraj**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
1. Automatic fetching of notifications while using the `useNotification` hook.
2. Stable referential identity for `refetch` method
- **What is the current behavior?** (You can also link to an open issue here)
1. users have to manually invoke `refetch` on load to get the notification
2. adding `refetch` as a dependency results in infinite loop
- **What is the new behavior (if this is a feature change)?**
1. Automatic fetching of notifications while using the `useNotification` hook.
2. Stable referential identity for `refetch` method
- **Other information**:

### CODE REVIEW

File: packages/notification-center/src/hooks/use-notifications.hook.ts [hunk 1]
```diff
@@ -37,9 +37,13 @@ export function useNotifications(props?: IUseNotificationsProps) {
     await mapUpdateAction(messageId, actionButtonType, status, payload, storeId);
   }
 
-  async function refetch() {
+  const refetch = useCallback(async () => {
```
Comment (scopsy) [inline packages/notification-center/src/hooks/use-notifications.hook.ts:40]: @BiswaViraj do we also need to change our notification center implementation?

File: packages/notification-center/src/hooks/use-notifications.hook.ts [hunk 2]
```diff
@@ -37,9 +37,13 @@ export function useNotifications(props?: IUseNotificationsProps) {
...
     await mapRefetch(storeId);
-  }
+  }, []);
+
+  useEffect(() => {
+    refetch();
+  }, [refetch]);
```
Comment (scopsy) [inline packages/notification-center/src/hooks/use-notifications.hook.ts:46]: A small question, why do we need this dependency here? Can't we just a call refetch on component mount?

Comment (BiswaViraj) [reply packages/notification-center/src/hooks/use-notifications.hook.ts:46]: we don't need the dependency here, but it doesn't hurt to have it either

File: packages/notification-center/src/hooks/use-notifications.hook.ts (hunk 1 above)
Comment (BiswaViraj) [reply packages/notification-center/src/hooks/use-notifications.hook.ts:40]: Like on the Admin panel?
I guess not, coz it's only for the scenarios where `useNotification` hook is used, and we are not using any on the web

**Comment by scopsy** [comment]
Was resolved in another PR when refactored the notification center providers dependency. So will close for now 🙏

**Pull request closed without merging**